types/marshalling.go

```diff
@@ -0,0 +1,147 @@
+package types
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+// TxPosition holds the start and end indexes (in the overall encoded []byte)
+// for a given txs field.
```

Comment:
is this the position of a single transaction or all the transactions?

Reply:
single transaction